pkg/operator/snapshotcrd/controller_test.go

```diff
@@ -0,0 +1,277 @@
+			OperatorSpec: opv1.OperatorSpec{
+				ManagementState: opv1.Managed,
+			},
+		},
+		Status: opv1.StorageStatus{},
+	}
+	for _, modifier := range modifiers {
+		cr = modifier(cr)
+	}
+	return cr
+}
+
+func withTrueConditions(conditions ...string) crModifier {
+	return func(i *opv1.Storage) *opv1.Storage {
+		if i.Status.Conditions == nil {
+			i.Status.Conditions = []opv1.OperatorCondition{}
+		}
+		for _, c := range conditions {
+			i.Status.Conditions = append(i.Status.Conditions, opv1.OperatorCondition{
+				Type:   c,
+				Status: opv1.ConditionTrue,
+			})
+		}
+		return i
+	}
+}
+
+func withFalseConditions(conditions ...string) crModifier {
+	return func(i *opv1.Storage) *opv1.Storage {
+		if i.Status.Conditions == nil {
+			i.Status.Conditions = []opv1.OperatorCondition{}
+		}
+		for _, c := range conditions {
+			i.Status.Conditions = append(i.Status.Conditions, opv1.OperatorCondition{
+				Type:   c,
+				Status: opv1.ConditionFalse,
+			})
+		}
+		return i
+	}
+}
+
+func TestSync(t *testing.T) {
+	tests := []operatorTest{
+		{
+			name: "no CRDs",
+			initialObjects: testObjects{
+				storage: getCR(),
+			},
+			expectedObjects: testObjects{
+				storage: getCR(
+					withTrueConditions(conditionsPrefix + opv1.OperatorStatusTypeUpgradeable),
+				),
+			},
+			expectErr: false,
+		},
+		{
+			name: "beta CRDs",
+			initialObjects: testObjects{
+				storage: getCR(),
+				crds: []*apiextv1.CustomResourceDefinition{
+					getCRD("volumesnapshots.snapshot.storage.k8s.io", "v1beta1"),
+					getCRD("volumesnapshotcontents.snapshot.storage.k8s.io", "v1beta1"),
+					getCRD("volumesnapshotclassess.snapshot.storage.k8s.io", "v1beta1"),
+				},
+			},
+			expectedObjects: testObjects{
+				storage: getCR(
+					withTrueConditions(conditionsPrefix + opv1.OperatorStatusTypeUpgradeable),
+				),
+			},
+			expectErr: false,
+		},
+		{
+			name: "alpha CRDs",
+			initialObjects: testObjects{
+				storage: getCR(),
+				crds: []*apiextv1.CustomResourceDefinition{
+					getCRD("volumesnapshots.snapshot.storage.k8s.io", "v1alpha1"),
+					getCRD("volumesnapshotcontents.snapshot.storage.k8s.io", "v1alpha1"),
+					getCRD("volumesnapshotclassess.snapshot.storage.k8s.io", "v1alpha1"),
+				},
+			},
+			expectedObjects: testObjects{
+				storage: getCR(
+					withFalseConditions(conditionsPrefix + opv1.OperatorStatusTypeUpgradeable),
+				),
+			},
+			expectErr: false,
+		},
+		{
+			name: "mixed CRDs",
+			initialObjects: testObjects{
+				storage: getCR(),
+				crds: []*apiextv1.CustomResourceDefinition{
+					getCRD("volumesnapshots.snapshot.storage.k8s.io", "v1alpha1", "v1beta1"),
+					getCRD("volumesnapshotcontents.snapshot.storage.k8s.io", "v1beta1"),
+					getCRD("volumesnapshotclassess.snapshot.storage.k8s.io", "v1alpha1"),
+				},
+			},
+			expectedObjects: testObjects{
+				storage: getCR(
+					withFalseConditions(conditionsPrefix + opv1.OperatorStatusTypeUpgradeable),
+				),
+			},
+			expectErr: false,
+		},
```

Comment:
Can we add test for case where there were other conditions on the CR and they should not be replaced when Upgradable condition is changed?

Reply:
Test added